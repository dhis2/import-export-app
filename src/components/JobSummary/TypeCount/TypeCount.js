import i18n from '@dhis2/d2-i18n'
import {
    Table,
    TableBody,
    TableCell,
    TableHead,
    TableRow,
    TableRowHead,
    TableCellHead,
} from '@dhis2/ui'
import PropTypes from 'prop-types'
import React from 'react'
import { FormField } from '../../index.js'
import { statsPropTypeObj } from '../helper.js'

const TypeCount = ({ stats }) => {
    if (stats.length == 0) {
        return null
    }
    return (
        <FormField
            label={`${i18n.t('Details by type')}`}
            dataTest="job-summary-type-count"
            name="typeCount"
        >
            <Table>
                <TableHead>
                    <TableRowHead>
                        <TableCellHead>{i18n.t('Type')}</TableCellHead>
                        <TableCellHead>{i18n.t('Created')}</TableCellHead>
                        <TableCellHead>{i18n.t('Deleted')}</TableCellHead>
                        <TableCellHead>{i18n.t('Ignored')}</TableCellHead>
                        <TableCellHead>{i18n.t('Updated')}</TableCellHead>
                        <TableCellHead>{i18n.t('Total')}</TableCellHead>
                    </TableRowHead>
                </TableHead>
                <TableBody>
                    {stats.map((s, i) => (
                        <TableRow key={`job-summary-type-count-${i}`}>
                            <TableCell>{s.type}</TableCell>
                            <TableCell>{s.created}</TableCell>
                            <TableCell>{s.deleted}</TableCell>
                            <TableCell>{s.ignored}</TableCell>
                            <TableCell>{s.updated}</TableCell>
                            <TableCell>{s.total}</TableCell>
                        </TableRow>
                    ))}
                </TableBody>
            </Table>
        </FormField>
    )
}

const typeCountStatsPropTypeObj = {
    ...statsPropTypeObj,
    type: PropTypes.string.isRequired,
}

TypeCount.propTypes = {
    stats: PropTypes.arrayOf(PropTypes.shape(typeCountStatsPropTypeObj))
        .isRequired,
}

export { TypeCount, typeCountStatsPropTypeObj }
