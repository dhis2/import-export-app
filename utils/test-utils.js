import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
