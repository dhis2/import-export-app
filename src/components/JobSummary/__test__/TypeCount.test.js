import React from 'react'
import { render } from 'test-utils'
import '@testing-library/jest-dom/extend-expect.js'

import { TypeCount } from '../TypeCount/TypeCount.js'

const props = {
    stats: [
        {
            type: 'DefaultGmlImportService',
            created: 1,
            deleted: 2,
            ignored: 3,
            updated: 4,
            total: 10,
        },
        {
            type: 'SndGmlImportService',
            created: 1,
            deleted: 2,
            ignored: 3,
            updated: 4,
            total: 10,
        },
    ],
}

it(`matches snapshot`, () => {
    const { asFragment } = render(<TypeCount {...props} />)
    expect(asFragment()).toMatchSnapshot()
})
