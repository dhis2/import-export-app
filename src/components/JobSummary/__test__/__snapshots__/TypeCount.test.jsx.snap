// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="job-summary-type-count"
  >
    <span
      class="label"
    >
      Details by type
    </span>
    <table
      class="jsx-2430604489 "
      data-test="dhis2-uicore-table"
    >
      <thead
        data-test="dhis2-uicore-tablehead"
      >
        <tr
          class="jsx-3124144591 zebraStriping"
          data-test="dhis2-uicore-tablerowhead"
        >
          <th
            class="jsx-1796049005 "
            data-test="dhis2-uicore-tablecellhead"
          >
            Type
          </th>
          <th
            class="jsx-1796049005 "
            data-test="dhis2-uicore-tablecellhead"
          >
            Created
          </th>
          <th
            class="jsx-1796049005 "
            data-test="dhis2-uicore-tablecellhead"
          >
            Deleted
          </th>
          <th
            class="jsx-1796049005 "
            data-test="dhis2-uicore-tablecellhead"
          >
            Ignored
          </th>
          <th
            class="jsx-1796049005 "
            data-test="dhis2-uicore-tablecellhead"
          >
            Updated
          </th>
          <th
            class="jsx-1796049005 "
            data-test="dhis2-uicore-tablecellhead"
          >
            Total
          </th>
        </tr>
      </thead>
      <tbody
        data-test="dhis2-uicore-tablebody"
      >
        <tr
          class="jsx-3124144591 zebraStriping"
          data-test="dhis2-uicore-tablerow"
        >
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            DefaultGmlImportService
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            1
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            2
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            3
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            4
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            10
          </td>
        </tr>
        <tr
          class="jsx-3124144591 zebraStriping"
          data-test="dhis2-uicore-tablerow"
        >
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            SndGmlImportService
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            1
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            2
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            3
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            4
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            10
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
